test/main/store/actions/index.test.js

```diff
@@ -1236,3 +1237,45 @@ describe('#navClearReq', () => {
     expect(nav).toStrictEqual([])
   })
 })
+
+describe('#updateTypedDataRequest', () => {
+  let requests
+  const request = '79928538-c971-4cf0-8498-fa4e8017398b'
+
+  const updaterFn = (node, account, leaf, reqId, update) => {
+    expect(node).toBe('main.accounts')
+    expect(account).toBe(owner)
+    expect(leaf).toBe('requests')
+
+    requests[reqId] = update(requests[reqId])
+  }
+
+  const updateSignatureMessage = (reqId, newData) => updateTypedDataAction(updaterFn, owner, reqId, newData)
+
+  beforeEach(() => {
+    requests = {
+      [request]: {
+        handlerId: '79928538-c971-4cf0-8498-fa4e8017398b',
+        type: 'signTypedData',
+        typedMessage: {
+          data: {
+            oldAttribute: true
+          }
+        }
+      },
+      some_other_id: {
+        handlerId: 'wow_such_valid_handerId'
+      }
+    }
+  })
+
+  it('should update a typed data request ', () => {
+    expect(requests[request].typedMessage.data.oldAttribute).toBeTruthy()
+    updateSignatureMessage(request, {
+      newAttribute: true
+    })
+
+    expect(requests[request].typedMessage.data.oldAttribute).toBeFalsy()
```

Comment:
why is old attribute falsy now, does it just get removed? might want to separate this out into two separate tests to remove any confusion. also they have can have more specific names rather than "should update the request" which doesn't give a lot of insight into what is being tested specifically